common/flatpak-transaction.c

```diff
@@ -2018,10 +2060,32 @@ add_deps (FlatpakTransaction          *self,
 
   if (op->kind == FLATPAK_TRANSACTION_OPERATION_UNINSTALL)
     {
-      /* If the runtime this app uses is already to be uninstalled, then this uninstall must happen before
-         the runtime is uninstalled */
+      g_autoptr(GBytes) deploy_data = NULL;
+
+      /* If the runtime is EOL and unused and in the same dir, uninstall it too. */
+      if (runtime_op == NULL &&
+          !priv->disable_deps &&
+          dir_ref_is_installed (priv->dir, full_runtime_ref, &runtime_remote, &deploy_data))
+        {
+          gboolean runtime_unused;
+
+          if (flatpak_deploy_data_get_eol (deploy_data) == NULL &&
+              flatpak_deploy_data_get_eol_rebase (deploy_data) == NULL)
+            return TRUE;
+
+          if (!runtime_will_be_unused (priv->installation, full_runtime_ref, op->ref, &runtime_unused, error))
+            return FALSE;
+
+          if (runtime_unused)
+            runtime_op = flatpak_transaction_add_op (self, runtime_remote, full_runtime_ref, NULL, NULL, NULL, NULL,
+                                                     FLATPAK_TRANSACTION_OPERATION_UNINSTALL);
```

Comment:
What about further dependencies of the runtime, like the locale extension of it?

Reply:
This is fixed now